### PR TITLE
clientv3: handle 'https' scheme in endpoint

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
@heyitsanthony @xiang90 Currently passing `https://` scheme in endpoints doesn't work. Just returning `transport closing` errors, because we are setting `tcp` proto for all `http` and `https` schemes.

File: clientv3/balancer.go [hunk 1]
```diff
@@ -205,6 +205,14 @@ func (b *simpleBalancer) Close() error {
 	return nil
 }
 
+func getScheme(ep string) string {
```
Comment (heyitsanthony) [inline clientv3/balancer.go]: why not change `parseEndpoints` to return the scheme string instead of a bool instead of having a new function?

**Comment by gyuho** [comment]
@heyitsanthony @xiang90 All fixed. PTAL. Thanks!

**Comment by xiang90** [comment]
> Currently passing https:// scheme in endpoints doesn't work. 

Can we add a test for this?

**Comment by gyuho** [comment]
Sure will add some tests.

**Comment by gyuho** [comment]
@xiang90 PTAL.

The test would fail without this patch.

Thanks.

**Pull request merged**